internal/services/hsm/cloud_hardware_security_module_cluster_resource.go

```diff
@@ -0,0 +1,466 @@
+						Computed: true,
+					},
+					"group_ids": {
+						Type:     pluginsdk.TypeList,
+						Computed: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+						},
+					},
+					"private_endpoint": {
+						Type:     pluginsdk.TypeList,
+						Computed: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"id": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+							},
+						},
+					},
+					"private_link_service_connection_state": {
+						Type:     pluginsdk.TypeList,
+						Computed: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"status": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+								"description": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+								"actions_required": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		"status_message": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (r CloudHardwareSecurityModuleClusterResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 60 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.HSM.CloudHsmClustersClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model CloudHardwareSecurityModuleClusterModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			id := cloudhsmclusters.NewCloudHsmClusterID(subscriptionId, model.ResourceGroupName, model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			parameters := cloudhsmclusters.CloudHsmCluster{
+				Location: location.Normalize(model.Location),
+				Tags:     tags.Expand(model.Tags),
+				Sku: &cloudhsmclusters.CloudHsmClusterSku{
+					Name:   cloudhsmclusters.CloudHsmClusterSkuNameStandardBOne,
+					Family: cloudhsmclusters.CloudHsmClusterSkuFamilyB,
+					// Capacity is left unset
+				},
+			}
+
+			if len(model.Identity) > 0 {
+				expandedIdentity, err := identity.ExpandUserAssignedMapFromModel(model.Identity)
+				if err != nil {
+					return fmt.Errorf("expanding `identity`: %+v", err)
+				}
+				// Convert to LegacySystemAndUserAssignedMap since the SDK uses the legacy type
+				legacyIdentity := &identity.LegacySystemAndUserAssignedMap{
+					Type:        identity.TypeUserAssigned,
+					IdentityIds: expandedIdentity.IdentityIds,
+				}
+				parameters.Identity = legacyIdentity
+			}
+
+			if model.AutoGeneratedDomainNameLabelScope != "" {
```

Comment:
since `auto_generated_domain_name_label_scope` has a default value, it won't be empty?